client/txn.go

```diff
@@ -406,12 +406,21 @@ func (txn *Txn) send(calls ...proto.Call) error {
 		}
 	}
 
-	_, haveEndTxn := lastReq.(*proto.EndTransactionRequest)
+	endTxnRequest, haveEndTxn := lastReq.(*proto.EndTransactionRequest)
 	needEndTxn := txn.Proto.Writing || haveTxnWrite
+	elideEndTxn := haveEndTxn && !needEndTxn
 
-	if haveEndTxn && !needEndTxn {
+	if elideEndTxn {
 		calls = calls[:lastIndex]
 	}
 
-	return txn.db.send(calls...)
+	err := txn.db.send(calls...)
+	if elideEndTxn && err == nil {
+		if endTxnRequest.Commit {
```

Comment:
Add a comment here specifying that all close-out status in returned transaction must be set so it's clear to future maintainers that if something besides status is expected to be set, it's meant to be handled here for the case of elided end txn.

    // Manually set transaction status (and any other information typically set on
    // commit or abort based on success or failure of the batched calls.